src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -207,7 +207,7 @@ boolean existsMediaFileIn2StorageLocations(MediaEntity media,
             WHERE
             ((:externalObjectDirectoryQueryTypeEnumIndex=1 AND eod.media is not null AND eod.media = eod2.media) OR                 
             (:externalObjectDirectoryQueryTypeEnumIndex=2 
-            AND eod.transcriptionDocumentEntity=eod2.transcriptionDocumentEntity AND eod.annotationDocumentEntity=eod2.annotationDocumentEntity))
+            AND (eod.transcriptionDocumentEntity=eod2.transcriptionDocumentEntity OR eod.annotationDocumentEntity=eod2.annotationDocumentEntity)))
```

Comment:
You still need the not null for transcription document and annotation document as you have for media on line 208